src/proxy/outbound.rs

```diff
@@ -214,9 +214,9 @@ impl OutboundConnection {
 
         // Block calls to ztunnel directly, unless we are in "in-pod".
         // For in-pod, this isn't an issue and is useful: this allows things like prometheus scraping ztunnel.
-        if self.pi.cfg.proxy_mode == ProxyMode::Shared
+        // TODO is this actually wrong for ProxyMode::Dedicated? Seems like a valid case
```

Comment:
Dedicated now implies ztunnel IP == app IP, so I don't think this is relevant anymore (?)

Reply:
https://github.com/istio/ztunnel/pull/1197